verilog/analysis/checkers/suggest_parentheses_rule.cc

```diff
@@ -0,0 +1,85 @@
+// Copyright 2017-2020 The Verible Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "verilog/analysis/checkers/suggest_parentheses_rule.h"
+
+#include "absl/strings/str_cat.h"
+#include "common/analysis/citation.h"
+#include "verilog/analysis/lint_rule_registry.h"
+#include "verilog/CST/expression.h"
+
+namespace verilog {
+namespace analysis {
+
+VERILOG_REGISTER_LINT_RULE(SuggestParenthesesRule);
+
+using verible::GetStyleGuideCitation;
+using verible::LintRuleStatus;
+using verible::LintViolation;
+using verible::SyntaxTreeContext;
+using verible::SymbolTag;
+
+absl::string_view SuggestParenthesesRule::Name() {
+  return "suggest-parentheses";
+}
+const char SuggestParenthesesRule::kTopic[] = "parentheses";
+const char SuggestParenthesesRule::kMessage[] =
+    "Use parentheses around nested expressions";
```

Comment:
I would make this message more specific to the implemented check:

"Parenthesize condition expressions that appear in the true-clause of another condition expression."

As more checks are added, they will get their own specific messages.